src/main/scala/nl/biopet/utils/SemanticVersion.scala

```diff
@@ -39,31 +38,59 @@ case class SemanticVersion(major: Int,
     this.build == that.build
   }
 
+  /**
+    * Compare two Option[Int] values.
+    * @param x an Option[Int]
+    * @param y the other Option[Int]
+    * @return Comparator Int.
+    */
+  def compareOptionInt(x: Option[Int], y: Option[Int]): Int = {
+    (x, y) match {
+      case (Some(_), None)    => +1 //Some(number) is always better than None
+      case (None, Some(_))    => -1
+      case (Some(a), Some(b)) => a compare b
+      case (None, None)       => 0
+      case _ =>
+        throw new IllegalStateException(s"Unable to compare '$x' and '$y'")
+    }
+  }
+
   /**
     * Checks whether one version is later than another.
     * Versions without builds (no -alpha, -SNAPSHOT or -build123)
     * are assumed to be later than versions with builds.
     * Example 0.8.0-alpha < 0.8.0-beta < 0.8.0
-    * @param that
-    * @return
+    * @param that another SemanticVersion
+    * @return Comparator int
     */
   def compare(that: SemanticVersion): Int = {
-    val versionCompare = (this.major, this.minor, this.patch) compare (that.major, that.minor, that.patch)
```

Comment:
Is it not easier to do this:
```scala
val versionCompare = (this.major, this.minor.getOrElse(0), this.patch.getOrElse(0)) compare (that.major, that.minor.getOrElse(0), that.patch.getOrElse(0)) 
```

Reply:
Yes. Allthough I intuitively do not like substituting zero when the optional is not defined, I think that is in line with semantic versioning spec. Let's test